Userland/DevTools/HackStudio/LanguageServers/Cpp/CppComprehensionEngine.cpp

```diff
@@ -227,8 +227,18 @@ String CppComprehensionEngine::type_of_property(const DocumentData& document, co
     auto& parent = (const MemberExpression&)(*identifier.parent());
     auto properties = properties_of_type(document, type_of(document, *parent.m_object));
     for (auto& prop : properties) {
-        if (prop.name == identifier.m_name && prop.type->is_named_type())
-            return ((NamedType&)prop.type).m_name->full_name();
+        if (prop.name.name != identifier.m_name)
+            continue;
+        Type* type { nullptr };
+        if (prop.declaration->is_variable_declaration()) {
+            type = ((VariableDeclaration&)*prop.declaration).m_type.ptr();
+        }
+        if (!type)
+            continue;
+        if (!type->is_named_type())
+            continue;
+
+        return ((NamedType&)*type).m_name->full_name();
```

Comment:
It'd be really nice if these were either plain structs, or had getters instead of having `CppComprehensionEngine` reach into their private bits :P

(not really meant to be addressed in this PR, but it'd be nice to fix this eventually)